cpp/src/command_classes/SwitchMultilevel.cpp

```diff
@@ -166,70 +168,162 @@ bool SwitchMultilevel::HandleMsg
 	uint32 const _instance	// = 1
 )
 {
-	if( SwitchMultilevelCmd_Report == (SwitchMultilevelCmd)_data[0] )
-	{
-		Log::Write( LogLevel_Info, GetNodeId(), "Received SwitchMultiLevel report: level=%d", _data[1] );
-
-		if( ValueByte* value = static_cast<ValueByte*>( GetValue( _instance, SwitchMultilevelIndex_Level ) ) )
+	switch ((SwitchMultilevelCmd)_data[0])
+	{	
+		case SwitchMultilevelCmd_Report:
 		{
-			value->OnValueRefreshed( _data[1] );
-			value->Release();
+			Log::Write( LogLevel_Info, GetNodeId(), "Received SwitchMultiLevel report: level=%d", _data[1] );
+
+			if( ValueByte* value = static_cast<ValueByte*>( GetValue( _instance, SwitchMultilevelIndex_Level ) ) )
+			{
+				value->OnValueRefreshed( _data[1] );
+				value->Release();
+			}
+			return true;
 		}
-		return true;
-	}
 
-	if( SwitchMultilevelCmd_SupportedReport == (SwitchMultilevelCmd)_data[0] )
-	{
-		uint8 switchType1 = _data[1] & 0x1f;
-		uint8 switchType2 = _data[2] & 0x1f;
-		uint8 switchtype1label = switchType1;
-		uint8 switchtype2label = switchType2;
-		if (switchtype1label > 7) /* size of c_switchLabelsPos, c_switchLabelsNeg */
+		case SwitchMultilevelCmd_Set:
```

Comment:
I believe I added that while trying to get the bright/dim received events to work. Will remove...